libs/core/tests/unit_tests/runnables/test_context.py

```diff
@@ -0,0 +1,337 @@
+from typing import Any, Callable, List, NamedTuple, Union
+
+import pytest
+
+from langchain_core.output_parsers.string import StrOutputParser
+from langchain_core.prompt_values import StringPromptValue
+from langchain_core.prompts.prompt import PromptTemplate
+from langchain_core.runnables.base import Runnable, RunnableLambda
+from langchain_core.runnables.context import Context
+from langchain_core.runnables.passthrough import RunnablePassthrough
+from langchain_core.runnables.utils import aadd, add
+from tests.unit_tests.fake.llm import FakeListLLM, FakeStreamingListLLM
+
+
+class TestCase(NamedTuple):
+    input: Any
+    output: Any
+
+
+def seq_naive_rag() -> Runnable:
+    context = [
+        "Hi there!",
+        "How are you?",
+        "What's your name?",
+    ]
+
+    retriever = RunnableLambda(lambda x: context)
+    prompt = PromptTemplate.from_template("{context} {question}")
+    llm = FakeListLLM(responses=["hello"])
+
+    return (
+        Context.setter("input")
```

Comment:
i.e. rather than creating a setter or getter using `Context.setter` or `Context.getter`, use something like:

```python
context = Context.create(scope="my_scoped_local_context") # either explicitly pass a scope string or use randomly generated UUID (if not provided)

return context.setter("input") | ... | {
    "result": RunnablePassthrough(),
    "context": context.getter("context"),
    "input": context.getter("input"),
}
```

Reply:
+1 would be nice to be able to create a scope

Reply:
Added this as an option in the latest commit